provider-ci/internal/pkg/templates/bridged-provider/.github/workflows/upgrade-provider.yml

```
@@ -16,9 +16,15 @@ on:
     # 3 AM UTC ~ 8 PM PDT / 7 PM PST daily. Time chosen to run during off hours.
     - cron: 0 3 * * *
 
+permissions:
+  contents: write
+  issues: write
+  pull-requests: write
+
 env:
-  GH_TOKEN: ${{ secrets.PULUMI_BOT_TOKEN }}
+  GH_TOKEN: ${{ secrets.UPGRADE_PROVIDER_TOKEN }}
```

I think for each of these we need fallbacks, something like the following ... 

```suggestion
  GH_TOKEN: ${{ secrets.UPGRADE_PROVIDER_TOKEN || secrets.PULUMI_BOT_TOKEN || secrets.GITHUB_TOKEN }}
```

---

@danielrbradley does such a logical OR work within the interpolation?

---

From the point of view of the org-level secret, I'm not quite convinced this is the right name yet.

Perhaps something more like `PULUMI_PROVIDER_AUTOMATED_PR_TOKEN` .. though shorter would be better.

---

I've not tested the interpolation with ORs, but I _think_ it should work.

---

https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/evaluate-expressions-in-workflows-and-actions#example

```
env:
  MY_ENV_VAR: ${{ github.ref == 'refs/heads/main' && 'value_for_main_branch' || 'value_for_other_branches' }}
```

---

Suggestions:
- `PULUMI_PROVIDER_AUTO_PR_TOKEN`
- `PULUMI_PROVIDER_AUTOMATION_TOKEN`